src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -90,12 +88,15 @@ internal class CompletionHandler : ILspServiceDocumentRequestHandler<LSP.Complet
                 return null;
 
             var (list, isIncomplete, resultId) = completionListResult.Value;
-            if (list.IsEmpty)
+
+            if (list.ItemsList.Count == 0)
             {
                 return new LSP.VSInternalCompletionList
                 {
                     Items = Array.Empty<LSP.CompletionItem>(),
-                    SuggestionMode = list.SuggestionModeItem != null,
+                    // If we have a suggestion mode item, we just need to keep the list in suggestion mode.
+                    // We don't need to return the fake suggestion mode item.
+                    SuggestionMode = list.SuggestionModeItem is not null,
                     IsIncomplete = isIncomplete,
                 };
```

Comment:
so it's ok if there are no items in teh list and no suggestion item for us to return this guy?